src/com/epam/javatrainings/homework_6/Buffer.java

```diff
@@ -0,0 +1,40 @@
+package com.epam.javatrainings.homework_6;
+
+public class Buffer <T>{
+    private T obj;
+
+    public T getObj() {
+        return obj;
+    }
+
+    public void setObj(T obj) {
+        this.obj = obj;
+    }
+
+    public synchronized void  put(T obj) {
+        if (this.obj != null) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+            this.obj = obj;
+            notify();
+
+    }
+    public synchronized T take(){
+         if(this.obj == null){
```

Comment:
When the synchronized methods using by more than two threads then should be checked the shared object in while cycle, and call to notifyAll  method, because there is no guarantee that the notify method will wake up the expected thread.